ui/packages/consul-ui/app/components/data-source/index.js

```diff
@@ -89,6 +89,14 @@ export default class DataSource extends Component {
 
   @action
   disconnect() {
+    // FIXME? Should we be doing this here
+    if (
+      typeof this.data !== 'undefined' &&
+      typeof this.data.length === 'undefined' &&
+      typeof this.data.rollbackAttributes === 'function'
+    ) {
+      this.data.rollbackAttributes();
+    }
```

Comment:
If the data is a single item and it hasn't been saved at all, then roll it back when the DataSource is removed from the page.